owncloudApp/src/main/java/com/owncloud/android/operations/SyncProfileOperation.kt

```diff
@@ -60,30 +60,33 @@ class SyncProfileOperation(
                         setUserData(account, AccountUtils.Constants.KEY_ID, userInfo.id)
                     }
 
-                    val refreshUserQuotaFromServerAsyncUseCase: RefreshUserQuotaFromServerAsyncUseCase by inject()
-                    val userQuotaResult =
-                        refreshUserQuotaFromServerAsyncUseCase(
-                            RefreshUserQuotaFromServerAsyncUseCase.Params(
-                                account.name
-                            )
-                        )
-                    userQuotaResult.getDataOrNull()?.let {
-                        Timber.d("User quota synchronized for account ${account.name}")
+                    val getStoredCapabilitiesUseCase: GetStoredCapabilitiesUseCase by inject()
+                    val storedCapabilities = getStoredCapabilitiesUseCase(GetStoredCapabilitiesUseCase.Params(account.name))
 
-                        val getStoredCapabilitiesUseCase: GetStoredCapabilitiesUseCase by inject()
+                    storedCapabilities?.let {
+                        if (it.isSpacesAllowed()) {
+                            val refreshUserQuotaFromServerAsyncUseCase: RefreshUserQuotaFromServerAsyncUseCase by inject()
+                            val userQuotaResult =
+                                refreshUserQuotaFromServerAsyncUseCase(
+                                    RefreshUserQuotaFromServerAsyncUseCase.Params(
+                                        account.name
+                                    )
+                                )
+                            userQuotaResult.getDataOrNull()?.let {
+                                Timber.d("User quota synchronized for account ${account.name}")
+                                val shouldFetchAvatar = storedCapabilities?.isFetchingAvatarAllowed() ?: true
```

Comment:
Let's put the avatar stuff out of the `let` block of the capabilities. In case they are null, this won't be executed, and we want it to be executed as it was before

Reply:
Done 😃

Reply:
I meant, not only out of the user-quota `let` but out of the whole capabilities `let` 👍

Reply:
Ok, It's done